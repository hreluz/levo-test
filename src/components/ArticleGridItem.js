import PropTypes from 'prop-types';

const ArticleGridItem = ({date, title, paragraph, color = '#007bff'}) => {

    const titleSubSstring = title.length > 40 && `${ title.substring(0, 40)} ...` ;
    const paragraphSubSstring = paragraph.length > 300 && `${ paragraph.substring(0, 300)} ...` ;

    return (
        <>
        <div className="card text-white text-left p-3" style={{backgroundColor: `${color}` }}>
            <small className="text-left">{ date }</small>
            <blockquote className="blockquote mb-0">
                <p>{titleSubSstring}</p>
                <div className="blockquote-footer text-white">
                    <small>{paragraphSubSstring}</small>
                </div>
                <button type="button" className="btn btn-outline-light mt-4">Read More</button>
            </blockquote>
        </div>
        </>
    );
}

ArticleGridItem.propTypes = {
    date: PropTypes.string.isRequired,
    paragraph: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    color: PropTypes.string
}

export default ArticleGridItem;
