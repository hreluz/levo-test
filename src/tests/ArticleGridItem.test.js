import { shallow } from 'enzyme'
import ArticleGridItem from '../components/ArticleGridItem'

describe('Test in ArticleGridItem', () => {

    const title = 'A title';
    const paragraph = 'This is a paragraph to show for the test';
    const date = '24 December 2019';
    const wrapper = shallow(<ArticleGridItem
                                paragraph={paragraph}
                                date={date}
                                title={title}/>)

    test('it should show <ArticleGridItem/> correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('it should have a paragraph with the title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe( title );
    });

    test('it should have the date in the first small element', () => {
        const small = wrapper.find('small').at(0);
        expect(small.prop('children')).toBe(date);
    });

    test('it should have the paragraph in the second small element', () => {
        const small = wrapper.find('small').at(1);
        expect(small.prop('children')).toBe(paragraph);
    });
});