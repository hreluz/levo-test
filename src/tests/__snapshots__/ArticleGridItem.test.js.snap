// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in ArticleGridItem it should show <ArticleGridItem/> correctly 1`] = `
<Fragment>
  <div
    className="card text-white text-left p-3"
    style={
      Object {
        "backgroundColor": "#007bff",
      }
    }
  >
    <small>
      24 December 2019
    </small>
    <blockquote
      className="blockquote mb-0"
    >
      <p>
        A title
      </p>
      <div
        className="blockquote-footer text-white"
      >
        <small>
          This is a paragraph to show for the test
        </small>
      </div>
      <button
        className="btn btn-outline-light mt-4"
        type="button"
      >
        Read More
      </button>
    </blockquote>
  </div>
</Fragment>
`;
